src/main/java/logisticspipes/proxy/buildcraft/subproxies/LPBCPluggableState.java

```diff
@@ -28,17 +27,19 @@ public void readData(LPDataInputStream data) throws IOException {
     }
 
     @Override
-    @SneakyThrows({ IOException.class })
-    public boolean isDirty(boolean clean) {
-        LPDataOutputStream buffer = new LPDataOutputStream();
-        ByteBuf buf = Unpooled.buffer(128);
-        this.writeData(buf);
-        buffer.writeByteBuf(buf);
-        byte[] newBuffer = buffer.toByteArray();
-        boolean result = !Arrays.equals(newBuffer, oldBuffer);
-        if (clean && result) {
-            oldBuffer = newBuffer;
+    public synchronized boolean isDirty() {
```

Comment:
Hi! Thank you for PR!
Why do you need synchronized here?

Reply:
I'm not 100% sure if it's really needed. If two different threads call `isDirty` on the same `LPBCPluggableState` instance at the same time, they could write both to the same buffer. I saw that LP sets up a bunch of threads and also one for packet handling so I just played it safe.